client/src/app/pages/home/home.component.ts

```diff
@@ -50,6 +53,12 @@ export class HomeComponent implements OnInit {
       }
     });
   }
+
+  goToProfile(): void {  // click -> navigate to profile/username
+    this.router.navigate(['/profile/' + this.currentUser.username])
+    .then(() => this.storeFacade.loadUserTransactions(this.currentUser.id));
```

Comment:
we shouldn't load the transactions when the user navigates to the profile. We should load them when we load the user. The transactions will be updated with the frontend webhook. Let me know if you need help on where to call this

Reply:
hey @Alessandro100, I wanted to discuss this with you, actually there's a problem with how we use state right now. We are losing state info whenever we refresh or navigate to a new page. Let's say we navigate from home page to stock detail page, we need to load state user, stock etc, and if we don't load followers for example, we'll have a null for followers in the state.
I think we should presist the whole state all the time, maybe by using NgRx local storage?

Reply:
State does get lost when we refresh the page. This is why if a user logs in, we have their user id stored in local storage. Then when the page refreshes, it will check if there is an id in local storage -> if there is, we will GET the user. Now that the user is separated from their transactions and followers, we will need to make the same calls for those over there. In terms of losing state when navigating, that shouldn't be the case.

Reply:
Whats going on here on line 59 is everytime we go to the profile page, we will GET the transactions. This can lead to many unused calls